cpp/src/arrow/compute/kernels/scalar_random.cc

```diff
@@ -57,6 +57,14 @@ random::pcg64_oneseq MakeSeedGenerator() {
 }
 
 Status ExecRandom(KernelContext* ctx, const ExecSpan& batch, ExecResult* out) {
+  if (batch.IsNull()) {
```

Comment:
Since random is not a pure function, it should not return scalar like this.

Reply:
ok